pkg/container/network.go

```diff
@@ -53,35 +74,82 @@ func SetupContainerNetworking() ([]DHCPInterface, error) {
 	})
 
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return dhcpIfaces, nil
+	return fcIfaces, dhcpIfaces, nil
 }
 
-func networkSetup(dhcpIfaces *[]DHCPInterface) (bool, error) {
-	ifaces, err := net.Interfaces()
-	if err != nil || ifaces == nil || len(ifaces) == 0 {
+func filterIgnored(allIfaces []net.Interface, extraIntfs map[string]struct{}) (result []net.Interface) {
+
+	for _, intf := range allIfaces {
+
+		// first process explicitly ignored
+		if _, ok := ignoreInterfaces[intf.Name]; ok {
+			continue
+		}
+
+		// next process extra intfs
+		if _, ok := extraIntfs[intf.Name]; ok {
+			result = append(result, intf)
+		}
+
+		// add intfs with IPs
+		addrs, _ := intf.Addrs()
+		if len(addrs) > 0 {
+			result = append(result, intf)
+		}
+
+	}
+
+	return result
+}
+
+func networkSetup(fcIfaces *firecracker.NetworkInterfaces, dhcpIfaces *[]DHCPInterface, extraIntfs map[string]struct{}, expectedIntfNum *int) (bool, error) {
+	allIfaces, err := net.Interfaces()
+	if err != nil || allIfaces == nil || len(allIfaces) == 0 {
 		return true, fmt.Errorf("cannot get local network interfaces: %v", err)
 	}
 
-	// interfacesCount counts the interfaces that are relevant to Ignite (in other words, not ignored)
-	interfacesCount := 0
+	ifaces := filterIgnored(allIfaces, extraIntfs)
+	if len(ifaces) < *expectedIntfNum {
+		return true, fmt.Errorf("not enough extra interfaces connected (%d/%d), waiting", len(ifaces), *expectedIntfNum)
+	}
```

Comment:
If we add `vm` to the `networkSetup` function signature, we can get rid of all of the counting logic and the passing around of a single number.

It will be possible here to just loop through the interface names and compare them with what's in the netns.

This will clean up the code.
The error message will be able to include exactly which interface names are missing and what their expected modes(`tc-redirect`/`dhcp-bridge`/`macvtap`) are.